main.go

```diff
@@ -197,16 +200,25 @@ func main() {
 		// ``MINIO_PROFILER`` supported options are [cpu, mem, block].
 		switch os.Getenv("MINIO_PROFILER") {
```

Comment:
All you need is move it out of app.Before and place it above app.RunandExitOnError(). Previous defer style code is simpler.